torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1736,10 +1812,21 @@ def _lazy_init(self) -> None:
                 fsdp_module._is_root = False
                 fsdp_module._streams = self._streams
                 fsdp_module._exec_order_data = self._exec_order_data
+                if fsdp_module.limit_all_gathers != self.limit_all_gathers:
+                    # Prefer the root's value
+                    inconsistent_limit_all_gathers = True
+                    fsdp_module.limit_all_gathers = self.limit_all_gathers
+                fsdp_module._free_event_queue = self._free_event_queue
                 fsdp_module._handles_prefetched = self._handles_prefetched
                 fsdp_module._need_pre_backward_unshard = self._need_pre_backward_unshard
                 for handle in fsdp_module._handles:
                     fsdp_module._init_param_attributes(handle)
+        if inconsistent_limit_all_gathers:
```

Comment:
I guess it should be rarely happened? almost can not happen with autowrapping?

Reply:
Yes, this should not happen with auto wrapping, only with manual wrapping. @rohan-varma pointed the possibility of inconsistency out in the old version of the PR, and I think it is good to have as a safety.